src/frontend/vite.config.ts

```diff
@@ -51,8 +52,36 @@ export default defineConfig({
   base: '',
   build: {
     manifest: true,
-    outDir: '../../src/backend/InvenTree/web/static/web',
-    sourcemap: is_coverage
+    // outDir: '../../src/backend/InvenTree/web/static/web',
+    // TODO: outDir should be set to the correct path,
+    // TODO: depending on whether we are building library or UI?
+    sourcemap: true, // is_coverage,
+    lib: {
+      entry: {
+        components: resolve(__dirname, 'lib/components.ts'),
+        core: resolve(__dirname, 'lib/core.ts'),
+        functions: resolve(__dirname, 'lib/functions.ts'),
+        tables: resolve(__dirname, 'lib/tables.ts')
+      },
+      name: 'InvenTree',
+      formats: ['es']
+    },
+    rollupOptions: {
+      // TODO: This should be updated for all extrenal deps
+      external: ['React'],
+      output: {
+        globals: {
```

Comment:
Does this really work that way? Not sure about versions. I would vote for letting the user install its own dependencies, and mark these all as peerdependencies in our package.json. 

E.g. dependency version mismatches between an installed inventree PUI version and the plugin's version of the inventree UI package. This will limit the supported inventree version of a plugin drastically.

Reply:
> Does this really work that way?

🤷 

What I *want* to happen is that we require the user to install their own deps (as you say), and not bundle any external large libs (such as react / mantine) into the distributed libs. I *think* that is what you are also saying here?